### DISCUSSION TITLE
rune type needs a .len method

### DISCUSSION THREAD

**Discussion by JalonSolov** [vlang/v#12700] [Ideas]
... so you can tell how many bytes it takes, since there's no easy way to iterate over a string with runes, or slice the string to remove a single rune, or...

**Comment by medvednikov** [comment]
> no easy way to iterate over a string with runes

```for r in str.runes() {``` ?

**Comment by JalonSolov** [comment]
I did go a little out there, and I was thinking more about slicing on runes to change the string, but... a `.len` would still be useful for other reasons.  If nothing else, it would be more consistent to have it.

**Reply by ntrel** [reply]
There is a method for this:
https://modules.vlang.io/#rune.length_in_bytes

**Reply by ntrel** [reply]
Added doc:
https://github.com/vlang/v/pull/13826/files

**Reply by JalonSolov** [reply]
Length in bytes is fine... useful in some cases, but it could still use `.len` to tell how many runes, not bytes, since that is the more logical use.

**Reply by ntrel** [reply]
Not sure what you mean, what would r.len do on a rune?

**Reply by JalonSolov** [reply]
Hmm... you're right, never mind. `rune` is a single code-point.  I was thinking of a string with runes in it, etc.

**Comment by changrui** [comment]
o, no, `[]rune.len` is working fine, and `rune` has `length_bytes` only.
